packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -344,7 +343,9 @@ export class Outbox {
 		// If so, do nothing, as pending state manager will resubmit it correctly on reconnect.
 		// Because flush() is a task that executes async (on clean stack), we can get here in disconnected state.
 		if (this.params.shouldSend()) {
-			const processedBatch = this.compressBatch(rawBatch, disableGroupedBatching);
+			const processedBatch = this.compressBatch(
+				shouldGroup ? rawBatch : this.params.groupingManager.groupBatch(rawBatch),
```

Comment:
Flipped?
```suggestion
				shouldGroup ? this.params.groupingManager.groupBatch(rawBatch) : rawBatch,
```

Reply:
Thanks for catching this. I added more testing to ensure this isn't broken. In this case, it would cause no batches to be grouped other than `blobAttach` batches, which would fail every time and block users